cmake/information_helpers.cmake

```diff
@@ -1,11 +1,98 @@
+function(filter_generator_expressions INPUT OUTPUT)
+    string(REGEX REPLACE ".+LINK:.+:(.+)>" "\\1"
+                    TMP "${INPUT}")
+    string(REGEX REPLACE ".+\\$<COMPILE_LANG_AND_ID:CUDA,NVIDIA>:.+>(.+)" "\\1"
+                    TMP "${TMP}")
+    string(REGEX REPLACE ".+INTERFACE:.+>" "" TMP "${TMP}")
+    set(${OUTPUT} "${TMP}" PARENT_SCOPE)
+endfunction()
+
+macro(ginkgo_interface_libraries_recursively INTERFACE_LIBS)
+    foreach(_libs ${INTERFACE_LIBS})
+        if (TARGET ${_libs})
+            if (upper_CMAKE_BUILD_TYPE STREQUAL "DEBUG" AND "${_libs}" MATCHES "ginkgo.*")
+                set(GINKGO_INTERFACE_LIB_NAME "-l${_libs}${CMAKE_DEBUG_POSTFIX}")
+            elseif("${_libs}" MATCHES "ginkgo.*") # Ginkgo libs are appended in the form -l
+                set(GINKGO_INTERFACE_LIB_NAME "-l${_libs}")
+            endif()
+
+            # Get the link flags and treat them
+            get_target_property(GINKGO_INTERFACE_LIBS_LINK_FLAGS "${_libs}"
+                INTERFACE_LINK_OPTIONS)
+            if (GINKGO_INTERFACE_LIBS_LINK_FLAGS)
```

Comment:
need to treat [SHELL:](https://cmake.org/cmake/help/latest/prop_tgt/LINK_OPTIONS.html?highlight=shell#option-de-duplication) correctly. I see `SHELL:-pthread` in my output